### DISCUSSION TITLE
Setting an error in client response

### DISCUSSION THREAD

**Discussion by kenfinnigan** [open-telemetry/opentelemetry-java#3054] [Q&A]
When making a request from within a service to another external service I create a child span.

On receiving the response, I was looking at ways to set an error onto the span. I found that I couldn't use `recordException()` as I have an HTTP Response and not a Java Exception. The best I could figure is:

```
clientSpan.setStatus(StatusCode.ERROR, responseContext.getStatusInfo().getReasonPhrase());
```

Which sets the HTTP Response status reason onto the span status.

Is that sufficient? Am I missing another way to do this?

I was wondering if would be nice to be able to set an exception event onto the span even when there isn't an actual exception?

**Comment by anuraaga** [comment]
@kenfinnigan That looks basically correct - exceptions are really for things with a stack trace, while whether a span is an error or not is the `StatusCode`. Actually it's ok to record exceptions for `StatusCode.OK` spans too, backends will interpret this differently but should generally consider them handled exceptions or similar.

However, you'll want to leave out the reason phrase generally. If you're instrumenting an HTTP client, you should keep the semantic conventions in mind

https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/http.md
https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/rpc.md

For example, HTTP Status is defined as ERROR for certain codes, and description should be left out if it just corresponds with the HTTP status code, filled as a separate attribute.

And we're working on an instrumentation API to capture these semantic-convention concepts to make it easier to write instrumentation, when it's ready you'll probably want to use it, in the meantime it might give some hints

https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/2596

For example, we have this

https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/2596/files#diff-7e37e82c23f60b30b877c2e8f6d3369fd1c18b01b60dcf0030b589259b445433

and we'd expect all HTTP instrumentations to use that and not worry about reimplementing the semantic conventions.

**Comment by Oberon00** [comment]
> it's ok to record exceptions for StatusCode.OK spans too, backends will interpret this differently but should generally consider them handled exceptions or similar.

Handled vs unhandled is determined by the `exception.escaped` attribute, according to our semantic conventions.
